### PR TITLE
handle tentacles config interface

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
tentacles updated in https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/63 (required for tests)
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: config/__init__.py [hunk 1]
```diff
@@ -299,11 +302,13 @@
...
 TENTACLE_DEFAULT_CONFIG = "default_config"
+TENTACLE_DEFAULT_FOLDER = "Default"
 
 # Web interface
 UPDATED_CONFIG_SEPARATOR = "_"
 GLOBAL_CONFIG_KEY = "global_config"
 EVALUATOR_CONFIG_KEY = "evaluator_config"
+DEACTIVATE_OTHERS = "deactivate_others"
 TRADING_CONFIG_KEY = "trading_config"
 COIN_MARKET_CAP_CURRENCIES_LIST_URL = "https://api.coinmarketcap.com/v2/listings/"
```
Comment (Herklos) [inline config/__init__.py]: In an other PR we should implement internationalization :earth_africa: :earth_americas:  :earth_asia:

File: evaluator/abstract_evaluator.py [hunk 2]
```diff
@@ -73,11 +81,25 @@ def get_config_file_error_message(cls, error):
             f". (error: {error})"
 
     @classmethod
-    def get_evaluator_config(cls):
+    def get_specific_config(cls, raise_exception=True, raw_file=False):
+        try:
+            if raw_file:
+                with open(cls.get_config_file_name()) as file:
+                    return file.read()
```
Comment (Herklos) [inline evaluator/abstract_evaluator.py]: Where are raw files used ?

File: interfaces/web/models/configuration.py [hunk 3]
```diff
@@ -64,72 +90,134 @@ def is_trading_persistence_activated():
...
+    import trading.trader.modes as modes
+    yield modes, modes.AbstractTradingMode, TRADING_MODE_KEY
+    import evaluator.Strategies as strategies
+    yield strategies, strategies.StrategiesEvaluator, STRATEGY_KEY
+    import evaluator.TA as ta
+    yield ta, AbstractEvaluator, TA_EVALUATOR_KEY
+    import evaluator.Social as social
+    yield social, AbstractEvaluator, SOCIAL_EVALUATOR_KEY
+    import evaluator.RealTime as rt
+    yield rt, AbstractEvaluator, RT_EVALUATOR_KEY
```
Comment (Herklos) [inline interfaces/web/models/configuration.py]: :+1:

File: tools/__init__.py [hunk 4]
```diff
@@ -15,10 +15,12 @@
 #  License along with this library.
 
 from __future__ import print_function
+from distutils.version import LooseVersion
 import os
 import sys
 
 MIN_PYTHON_VERSION = (3, 7)
+MIN_TENTACLE_MANAGER_VERSION = "1.0.9"
```
Comment (Herklos) [inline tools/__init__.py]: :+1:

File: trading/trader/modes/abstract_trading_mode.py [hunk 5]
```diff
@@ -51,17 +51,38 @@ def __init__(self, config, exchange):
...
-        return f"{TENTACLES_PATH}/{TENTACLES_TRADING_PATH}/{TENTACLES_TRADING_MODE_PATH}/{EVALUATOR_CONFIG_FOLDER}/" \
-               f"{cls.get_name() + CONFIG_FILE_EXT}"
+        return f"{cls.get_config_folder()}/{cls.get_name() + CONFIG_FILE_EXT}"
+
+    @classmethod
+    def get_config_file_schema_name(cls):
+        return f"{cls.get_config_folder()}/{cls.get_name()}_{SCHEMA}{CONFIG_FILE_EXT}"
+
+    @classmethod
+    def get_specific_config(cls, raise_exception=True, raw_file=False):
```
Comment (Herklos) [inline trading/trader/modes/abstract_trading_mode.py]: Maybe these two function can be inherited from a new "Tentacle" abstract class ?

File: evaluator/abstract_evaluator.py (hunk 2 above)
Comment (GuillaumeDSM) [reply evaluator/abstract_evaluator.py]: they are required in this format to be parsed in web interface js: then passed to config editor

File: trading/trader/modes/abstract_trading_mode.py (hunk 5 above)
Comment (GuillaumeDSM) [reply trading/trader/modes/abstract_trading_mode.py]: Yep, that's also what I thought, I think it's better to leave it to another PR though

Comment (Herklos) [reply trading/trader/modes/abstract_trading_mode.py]: Yes !

File: evaluator/abstract_evaluator.py (hunk 2 above)
Comment (Herklos) [reply evaluator/abstract_evaluator.py]: Ok its the raw json directly parsed by the JS library

**Pull request merged**